devito/ops/node_factory.py

```diff
@@ -0,0 +1,63 @@
+from devito import Dimension, TimeFunction
+from devito.ops.utils import namespace
+from devito.symbolics import Macro, split_affine
+from devito.types import Indexed, Array
+
+
+class OPSNodeFactory():
+    """
+    Generate OPS nodes for building an OPS expression.
+
+    Examples
+    --------
+    >>> a = OPSNodeFactory()
+    >>> a.new_symbol('symbol_name')
+    symbol_name
+    """
+
+    def __init__(self):
+        self.ops_args = {}
+
+    def new_ops_arg(self, indexed):
+        """
+        Create an :class:`Indexed` node using OPS representation.
+        If the pair ops argument name and time dimension were alredy created,
+        it will return the stored value associated with this pair.
+
+        Parameters
+        ----------
+        indexed : :class:`Indexed`
+            Indexed object using devito representation.
+
+        Returns
+        -------
+        :class:`Indexed`
+            Indexed node using OPS representation.
+        """
+
+        # Build the ops argument identifier.
```

Comment:
(nitpicking) convention used throughout devito: never put a full stop at the end of comments